packages/git/src/browser/git-widget.tsx

```diff
@@ -447,25 +574,270 @@ export class GitWidget extends GitDiffWidget implements StatefulWidget {
+                    transitionProperty: 'height',
+                    transitionDuration: `${GitWidget.TRANSITION_TIME_MS}ms`,
+                    transitionTimingFunction: 'linear'
+                };
+        }
+    }
+
+    protected styleLastCommitOnTopMovingUp(transitionState: 'start' | 'transitioning'): React.CSSProperties {
+        return this.styleLastCommitOnTop(transitionState, -5, -40);
+    }
+
+    protected styleLastCommitOnTopMovingDown(transitionState: 'start' | 'transitioning'): React.CSSProperties {
+        return this.styleLastCommitOnTop(transitionState, -40, -5);
+    }
+
+    protected styleLastCommitOnBottomMovingUp(transitionState: 'start' | 'transitioning'): React.CSSProperties {
+        return this.styleLastCommitOnBottom(transitionState, -40, -5);
+    }
+
+    protected styleLastCommitOnBottomMovingDown(transitionState: 'start' | 'transitioning'): React.CSSProperties {
+        return this.styleLastCommitOnBottom(transitionState, -5, -40);
+    }
+
+    protected styleLastCommitOnTop(transitionState: 'start' | 'transitioning', startingTop: number, endingTop: number): React.CSSProperties {
+        const base = {
+            display: 'flex',
+            marginTop: 0,
+            marginBottom: 0,
+            paddingTop: 0,
+            paddingBottom: 0,
+            borderTop: 0,
+            borderBottom: 0
+        };
+
+        switch (transitionState) {
+            case 'start':
+                return {
+                    ...base,
+                    position: 'absolute',
+                    top: startingTop,
+                };
+            case 'transitioning':
+                return {
+                    ...base,
+                    position: 'absolute',
+                    top: endingTop,
+                    transitionProperty: 'top',
+                    transitionDuration: `${GitWidget.TRANSITION_TIME_MS}ms`,
+                    transitionTimingFunction: 'linear'
+                };
+        }
+    }
+
+    protected styleLastCommitOnBottom(transitionState: 'start' | 'transitioning', startingBottom: number, endingBottom: number): React.CSSProperties {
+        const base = {
+            display: 'flex',
+            marginTop: 0,
+            marginBottom: 0,
+            paddingTop: 0,
+            paddingBottom: 0,
+            borderTop: 0,
+            borderBottom: 0
+        };
+
+        switch (transitionState) {
+            case 'start':
+                return {
+                    ...base,
+                    position: 'absolute',
+                    bottom: startingBottom,
+                };
+            case 'transitioning':
+                return {
+                    ...base,
+                    position: 'absolute',
+                    bottom: endingBottom,
+                    transitionProperty: 'bottom',
+                    transitionDuration: `${GitWidget.TRANSITION_TIME_MS}ms`,
+                    transitionTimingFunction: 'linear'
+                };
```

Comment:
[minor]: I would add the `default` case and throw an `Error` with the `transitionState`. Thoughts?